### ISSUE TITLE
Draw interaction with stopClick has jumpy behaviour

### ISSUE THREAD

**Issue by EvertEt** [openlayers/openlayers#12161]
**Describe the bug**
We did get a report with latest dev version that when using stopClick and fast drawing, sometimes it becomes glitchy: the map jumps and the point is placed somewhere else. 
Sorry for the vague description but the screen recording should hopefully make it clearer.

https://user-images.githubusercontent.com/9197891/112857589-49ff2080-90b1-11eb-9303-2eecc9e2769b.mov

**To Reproduce**
1. Go to https://openlayers.org/en/latest/examples/draw-features.html
2. Fork and change to latest dev version (due to https://github.com/openlayers/openlayers/issues/12147)
3. Enable stopClick
4. Draw quickly

**Expected behavior**
No jumping

**Comment by ahocevar** [comment]
Unfortunately I am unable to reproduce this. Is there anything else you can tell us about the issue that would increase the chance for being able to reproduce?

**Comment by EvertEt** [comment]
Some details I know by heart, but I will ask for more:
- MacOS
- Brave Browser
- Controlled by mouse (no touchpad)

**Comment by OlivierKamers** [comment]
I have collected some more information that might be helpful. I added a log in `handleEvent` of `src/ol/interaction/Draw.js` (filtering out `POINTERMOVE` events).

This gives the following result when I'm able to trigger the issue:
![image](https://user-images.githubusercontent.com/5834741/112948722-b113e800-9138-11eb-89cf-e34b6ecb7eb1.png)

 It looks like I'm clicking and moving my mouse a bit too fast and triggering drag events in-between the `pointerdown` and `pointerup` events. It also happens when drawing points and the behaviour I see then is that once the glitch happens new points don't appear anymore when I click, and suddenly all appear together when I drag the map again.

**Comment by ahocevar** [comment]
And it really only happens when `stopClick` is set to `true`? I can only guess because I'm unable to reproduce, but maybe this change fixes the issue:
```diff
--- a/src/ol/MapBrowserEventHandler.js
+++ b/src/ol/MapBrowserEventHandler.js
@@ -210,6 +210,10 @@ class MapBrowserEventHandler extends Target {
     ) {
       this.emulateClick_(this.down_);
     }
+    if (newEvent.propagationStopped && this.clickTimeoutId_ !== undefined) {
+      clearTimeout(this.clickTimeoutId_);
+      this.clickTimeoutId_ = undefined;
+    }
 
     if (this.activePointers_ === 0) {
       this.dragListenerKeys_.forEach(unlistenByKey);
```

**Comment by OlivierKamers** [comment]
Yes I can only reproduce it when `stopClick` is set to `true`. 

I tried out the change you proposed but it doesn't seem to change it. I added a log inside the if-case and it's not being triggered.

**Comment by ahocevar** [comment]
Ok, in this case I'm out of ideas. If you are able to find a fix, it would be much appreciated.

**Comment by EvertEt** [comment]
@OlivierKamers Since you could trigger this behaviour, could you check if it is fixed by #12185 or any different? Thank you!

**Comment by OlivierKamers** [comment]
@ahocevar #12185 seems to fix this issue! I couldn't reproduce it anymore with that change. The behaviour I get now is the same as before and the one that can be expected: the map is dragged a tiny bit instead of the drawing jumping around.
